### PR TITLE
Fix exception type in sesolve

### PR DESCRIPTION
**Pull request by eendebakpt**
**Description**

Return `TypeError` in `sesolve` when the argument `H` is of incorrect type.

### CODE REVIEW

File: doc/changes/1924.misc.py [hunk 1]
```diff
@@ -0,0 +1 @@
+Return TypeError instead of Exception for type error in sesolve argument
```
Comment (hodgestar) [inline doc/changes/1924.misc.py:1]: Thank you for adding the changelog with towncrier -- I think you are officially the first person to use this (after the PR that added it).

I think the filename should be just "1924.misc" though, without the `.py`.

File: qutip/tests/test_sesolve.py [hunk 2]
```diff
@@ -352,3 +352,10 @@ def test_sesolve_bad_e_ops():
     tlist = np.linspace(0, 20, 200)
     with pytest.raises(TypeError) as exc:
         sesolve(H, psi0, tlist=tlist, e_ops=[qeye(3)])
+
+def test_sesolve_bad_H():
+    H = sigmaz(),
+    psi0 = basis(2, 0)
+    tlist = np.linspace(0, 20, 200)
+    with pytest.raises(TypeError) as exc:
+        sesolve(H, psi0, tlist=tlist, e_ops=[qeye(3)])
```
Comment (hodgestar) [inline qutip/tests/test_sesolve.py:361]:
```suggestion
        sesolve(H, psi0, tlist=tlist, e_ops=[qeye(3)])
    assert str(exc.value).startswith("Invalid H:")
```
I guess we should check that the right exception message is raised too.

**Review by hodgestar — CHANGES_REQUESTED** [review]
Thank you for making the suggested change and adding the changelog. I was going to offer to make the changelog for you, but you beat me to it.

File: doc/changes/1924.misc.py (hunk 1 above)
Comment (eendebakpt) [reply doc/changes/1924.misc.py]: I was already wondering why the `changes` folder was so empty :-)

**Review by hodgestar — APPROVED** [review]
Approved! Will merge once the CI runs complete successfully.

**Pull request merged**